### PR TITLE
Add IdentityX data layer events

### PR DESCRIPTION
**Pull request by B77Mills**
Add push events to the dataLayerIdentityX for login, profile view & profile submit.  This also allows for passing custom event names into the given events.

### CODE REVIEW

File: packages/marko-web-identity-x/browser/profile.vue [hunk 1]
```diff
@@ -234,6 +238,18 @@ export default {
       type: Array,
       default: () => [],
     },
+    eventName: {
+      type: String,
+      default: 'profile',
+    },
```
Comment (solocommand) [inline packages/marko-web-identity-x/browser/profile.vue]: `eventName` isn't used within this component, only `viewEventName` and `submitEventName`.

File: packages/marko-web-identity-x/utils/gtm-events.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  if (dataLayer) dataLayer.push({ event, email });
+};
+
+export const profileView = ({ event = 'profile_view' }) => {
+  console.log('gtm-event-profile-view: ', event);
+  const dataLayer = window.dataLayerIdentityX;
+  if (dataLayer) dataLayer.push({ event });
+};
+
+export const profileSubmit = ({ event = 'update_profile', email }) => {
```
Comment (solocommand) [inline packages/marko-web-identity-x/utils/gtm-events.js]: You made the default value in the Vue component `profile_submit`, but the default value here is `update_profile`. Which is correct?

File: packages/marko-web-identity-x/utils/gtm-events.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable no-undef */
+export const loginSubmit = ({ event = 'login', email }) => {
+  console.log('gtm-event-login-submit: ', event);
```
Comment (solocommand) [inline packages/marko-web-identity-x/utils/gtm-events.js]: Remove logs

File: packages/marko-web-identity-x/components/form-profile.marko [hunk 4]
```diff
@@ -12,6 +12,9 @@ $ const { identityX } = req;
       hiddenFields: identityX.config.getHiddenFields(),
       callToAction: input.callToAction,
       reloadPageOnSubmit: input.reloadPageOnSubmit,
+      viewEventName: input.viewEventName,
+      submitEventName: input.submitEventName,
+      reloadPageOnSubmit: input.reloadPageOnSubmit,
```
Comment (solocommand) [inline packages/marko-web-identity-x/components/form-profile.marko]: `reloadPageOnSubmit` is now duplicated in the component props

**Review by solocommand — CHANGES_REQUESTED** [review]
A few changes. Be careful of implementation on the GTM side -- the email address **is** considered PII. Ultimately it's up to the implementor to ensure they are in compliance. Is there a specific request for the email address to be sent, or would the IdX user id suffice?

File: packages/marko-web-identity-x/browser/profile.vue (hunk 1 above)
Comment (B77Mills) [reply packages/marko-web-identity-x/browser/profile.vue]: I removed the eventName prop that was never used instead handle with the viewEventName & submitEventName props

**Review by solocommand — DISMISSED** [review]
@B77Mills looks good, but there's a conflict in the marko component definitions

**Comment by B77Mills** [comment]
@solocommand I removed the passing of email address on login completely and added the passing of userId instead of email on profile is submit.

**Comment by solocommand** [comment]
@brandonbk @B77Mills updated to include event dispatching. Notable changes:
- pass `eventLabel` instead of `eventName` to adjust the reported property
- Comment post view does not emit an event (noisy)
- There is still a race condition where events on pages with client-side redirects (authenticate, logout) may not consistently report

**Review by solocommand — APPROVED** [review]

**Pull request merged**